### PR TITLE
DAOS-13483 client: to fix libpil4dfs compiling issue on CentOS 7.9

### PR DESCRIPTION
**Pull request by wiliamhuang**
We do need to run DAOS benchmark on Frontera at TACC which runs CentOS 7.9. It would be helpful to let libpil4dfs support CentOS 7.9.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by wangshilong — APPROVED** [review]
Looks good, but better to verify on centos7 testing.

**Review by daltonbohning — APPROVED** [review]
Successfully compiles on Frontera, which is on EL7

**Review by jolivier23 — APPROVED** [review]

File: src/client/dfuse/pil4dfs/int_dfs.c [hunk 1]
```diff
@@ -1558,13 +1603,8 @@ query_fd_forward_dest(int fd_src)
...
 {
-	struct timespec times_loc;
-	char            file_name[128];
-
-	clock_gettime(CLOCK_REALTIME, &times_loc);
-	snprintf(file_name, sizeof(file_name) - 1, "dummy_%ld_%ld", times_loc.tv_sec,
-		 times_loc.tv_nsec);
-	return memfd_create(file_name, 0);
+	/* Just use open() to allocate a fd from kernel. Not going to read the file. */
+	return open("/proc/self/maps", O_RDONLY);
```
Comment (ashleypittman) [inline src/client/dfuse/pil4dfs/int_dfs.c:1607]: This looks like regression so perhaps a compile check would be better here, although looking at it this is only called in dup2 so perhaps this is good enough.

File: src/client/dfuse/pil4dfs/int_dfs.c [hunk 2]
```diff
@@ -5034,7 +5085,7 @@ sig_handler(int code, siginfo_t *siginfo, void *ctx)
 	int                     rc, fd, idx_map;
 	d_iov_t                 iov;
 	d_sg_list_t             sgl;
-	struct ucontext_t      *context = (struct ucontext_t *)ctx;
+	ucontext_t             *context = (ucontext_t *)ctx;
```
Comment (ashleypittman) [inline src/client/dfuse/pil4dfs/int_dfs.c:5088]: Technically you don't need this cast here as void * aliases to any pointer but this is a very common mistake.
```suggestion
	ucontext_t             *context = ctx;
```

**Comment by wiliamhuang** [comment]
All CI tests passed. @jolivier23 Can we land? Thank you!

**Comment by wiliamhuang** [comment]
@ashleypittman @jolivier23 Can we land this PR or do we need more changes? Thank you!

**Comment by ashleypittman** [comment]
> @ashleypittman @jolivier23 Can we land this PR or do we need more changes? Thank you!

when a PR is ready then the correct procedure is to add daos-gatekeeper as a reviewer, then it'll show up to the gatekeeping team.

**Comment by jolivier23** [comment]
> > > @ashleypittman @jolivier23 Can we land this PR or do we need more changes? Thank you!
> > 
> > 
> > when a PR is ready then the correct procedure is to add daos-gatekeeper as a reviewer, then it'll show up to the gatekeeping team.
> 
> I did this a couple of times in this PR alredy. :)

Every time a gatekeeper comments, it gets cleared.

**Pull request merged**